### PR TITLE
Legend position fix

### PR DESCRIPTION
**Pull request by mattdonut**
Plotly has some odd behavior in positioning the legened.

### CODE REVIEW

File: src/components/legend/draw.js [hunk 1]
```diff
@@ -661,7 +667,8 @@ function computeLegendDimensions(gd, groups, traces) {
 
 function expandMargin(gd) {
     var fullLayout = gd._fullLayout,
-        opts = fullLayout.legend;
+        opts = fullLayout.legend,
+        gs = fullLayout._size;
```
Comment (mattdonut) [inline src/components/legend/draw.js]: This isn't needed

File: src/components/legend/attributes.js [hunk 2]
```diff
@@ -109,5 +109,10 @@ module.exports = {
             'This anchor binds the `y` position to the *top*, *middle*',
             'or *bottom* of the legend.'
         ].join(' ')
+    },
+    maxHeight: {
+        valType: 'number',
+        role: 'style',
+        description: 'Sets the maximum allowed height (in pixels) of the legend'
```
Comment (eleland) [inline src/components/legend/attributes.js]: nit: need a period at the end of the description sentence to match others.

File: src/components/legend/draw.js [hunk 3]
```diff
@@ -135,7 +135,13 @@ module.exports = function draw(gd) {
 
     computeLegendDimensions(gd, groups, traces);
 
-    if(opts.height > lyMax) {
+    // The height that the legend will eventually be clipped too
+    var displayHeight = opts.height
```
Comment (eleland) [inline src/components/legend/draw.js]: nit: need semicolon.

**Review by eleland — COMMENTED** [review]
Looking good, just 2 minor nits.

**Pull request merged**